### ISSUE TITLE
very urgent: meschentral server doesn't start anymore

### ISSUE THREAD

**Issue by tradexsrl** [Ylianst/MeshCentral#3522]
Hi , came back this morning in the office and i discover that service of meshcentral is ko
if i try to run manually the service i get: 
![immagine](https://user-images.githubusercontent.com/62424179/150743159-88cab36e-96ef-4a18-a722-21d67030e0fc.png)
maybe a new release that automatically installed that crashed the server?
server is installed on windows 10 pro with latest nodejs
please let me know

**Comment by JSuenram** [comment]
Please post your config file and also you can run service from command-line with --debug. This will help to determine what is causing your issue.

**Comment by tradexsrl** [comment]
this is the output with --debug:
![immagine](https://user-images.githubusercontent.com/62424179/150744402-4d9c0efe-1126-443e-99fe-b6d5bef6969d.png)

**Comment by tradexsrl** [comment]
this is the config:
{
   "settings": {
      "cert": "xxxxx",
      "redirport": 80,
      "SessionTime": 30,	  
      "wanonly": true, 
      "_minify": true, 
      "selfupdate": true, 
      "clickonce": true,
      "AllowHighQualityDesktop": true, 
      "webrtc": true,
      "AgentPing": 50      
   }, 
   "domains": {
      "": {
         "footer": "\u003ca href=\"http://xxxxxxxxx\"\u003exxxxxx\u003c/a\u003e",
		 "UserQuota": 1048760000,
		 "MeshQuota": 1048760000,
		 "UserSessionIdleTimeout": 25
      }
   }, 
   "smtp": {
      "host": "x.x.x.x", 
      "port": 25, 
      "from": "x@x.com", 
      "user": "y@x.com", 
      "pass": "1234", 
      "tls": false
   },
   "letsencrypt": {
      "email": "x@x.com",
      "names": "xxxxxxxxx",
      "rsaKeySize": 3072,
      "production": true
   }
}

**Comment by JSuenram** [comment]
> 

You need to start the "exe" direct from command line, not the service to get debug info.

**Comment by tradexsrl** [comment]
i got also the log that report this:
C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js:177
    child.send('shutdown');
          ^

TypeError: Cannot read property 'send' of null
    at process.killkid (C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js:177:11)
    at process.emit (events.js:197:13)
    at process.exit (internal/process/per_thread.js:150:15)
    at monitor (C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js:99:17)
    at ChildProcess.<anonymous> (C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js:170:5)
    at ChildProcess.emit (events.js:197:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
(node:3784) ExperimentalWarning: queueMicrotask() is experimental.
(node:3764) ExperimentalWarning: queueMicrotask() is experimental.
(node:3928) ExperimentalWarning: queueMicrotask() is experimental.
(node:3716) ExperimentalWarning: queueMicrotask() is experimental.
(node:3732) ExperimentalWarning: queueMicrotask() is experimental.
(node:4284) ExperimentalWarning: queueMicrotask() is experimental.
(node:3740) ExperimentalWarning: queueMicrotask() is experimental.
(node:4420) ExperimentalWarning: queueMicrotask() is experimental.
(node:4024) ExperimentalWarning: queueMicrotask() is experimental.
(node:3808) ExperimentalWarning: queueMicrotask() is experimental.
(node:3840) ExperimentalWarning: queueMicrotask() is experimental.
(node:3720) ExperimentalWarning: queueMicrotask() is experimental.
(node:3696) ExperimentalWarning: queueMicrotask() is experimental.
(node:4672) ExperimentalWarning: queueMicrotask() is experimental.
(node:3548) ExperimentalWarning: queueMicrotask() is experimental.
(node:3684) ExperimentalWarning: queueMicrotask() is experimental.
(node:3776) ExperimentalWarning: queueMicrotask() is experimental.
(node:3736) ExperimentalWarning: queueMicrotask() is experimental.
(node:3680) ExperimentalWarning: queueMicrotask() is experimental.
(node:4000) ExperimentalWarning: queueMicrotask() is experimental.
(node:3872) ExperimentalWarning: queueMicrotask() is experimental.
(node:3588) ExperimentalWarning: queueMicrotask() is experimental.
(node:3712) ExperimentalWarning: queueMicrotask() is experimental.
(node:3804) ExperimentalWarning: queueMicrotask() is experimental.
(node:3876) ExperimentalWarning: queueMicrotask() is experimental.
(node:3856) ExperimentalWarning: queueMicrotask() is experimental.
(node:3856) ExperimentalWarning: queueMicrotask() is experimental.
(node:3844) ExperimentalWarning: queueMicrotask() is experimental.
(node:3608) ExperimentalWarning: queueMicrotask() is experimental.
(node:3744) ExperimentalWarning: queueMicrotask() is experimental.
(node:3836) ExperimentalWarning: queueMicrotask() is experimental.
(node:3796) ExperimentalWarning: queueMicrotask() is experimental.
(node:3736) ExperimentalWarning: queueMicrotask() is experimental.
(node:3620) ExperimentalWarning: queueMicrotask() is experimental.
(node:3860) ExperimentalWarning: queueMicrotask() is experimental.
(node:3716) ExperimentalWarning: queueMicrotask() is experimental.
(node:3620) ExperimentalWarning: queueMicrotask() is experimental.
(node:3732) ExperimentalWarning: queueMicrotask() is experimental.
(node:3728) ExperimentalWarning: queueMicrotask() is experimental.
(node:3816) ExperimentalWarning: queueMicrotask() is experimental.
(node:3320) ExperimentalWarning: queueMicrotask() is experimental.
(node:3752) ExperimentalWarning: queueMicrotask() is experimental.
(node:3636) ExperimentalWarning: queueMicrotask() is experimental.
(node:3840) ExperimentalWarning: queueMicrotask() is experimental.
(node:3784) ExperimentalWarning: queueMicrotask() is experimental.
(node:3856) ExperimentalWarning: queueMicrotask() is experimental.
(node:3728) ExperimentalWarning: queueMicrotask() is experimental.
(node:3940) ExperimentalWarning: queueMicrotask() is experimental.
(node:3780) ExperimentalWarning: queueMicrotask() is experimental.
(node:3740) ExperimentalWarning: queueMicrotask() is experimental.
(node:3828) ExperimentalWarning: queueMicrotask() is experimental.
(node:3752) ExperimentalWarning: queueMicrotask() is experimental.
(node:3844) ExperimentalWarning: queueMicrotask() is experimental.
(node:3756) ExperimentalWarning: queueMicrotask() is experimental.
(node:3708) ExperimentalWarning: queueMicrotask() is experimental.
(node:3820) ExperimentalWarning: queueMicrotask() is experimental.
(node:3852) ExperimentalWarning: queueMicrotask() is experimental.
(node:3744) ExperimentalWarning: queueMicrotask() is experimental.
(node:3764) ExperimentalWarning: queueMicrotask() is experimental.
(node:3808) ExperimentalWarning: queueMicrotask() is experimental.
(node:3772) ExperimentalWarning: queueMicrotask() is experimental.
(node:3644) ExperimentalWarning: queueMicrotask() is experimental.
(node:3780) ExperimentalWarning: queueMicrotask() is experimental.
(node:3736) ExperimentalWarning: queueMicrotask() is experimental.
(node:3744) ExperimentalWarning: queueMicrotask() is experimental.
(node:3700) ExperimentalWarning: queueMicrotask() is experimental.
(node:3896) ExperimentalWarning: queueMicrotask() is experimental.
(node:3852) ExperimentalWarning: queueMicrotask() is experimental.
(node:3724) ExperimentalWarning: queueMicrotask() is experimental.
(node:3780) ExperimentalWarning: queueMicrotask() is experimental.
(node:3796) ExperimentalWarning: queueMicrotask() is experimental.
(node:3840) ExperimentalWarning: queueMicrotask() is experimental.
(node:3716) ExperimentalWarning: queueMicrotask() is experimental.
(node:3792) ExperimentalWarning: queueMicrotask() is experimental.
(node:3696) ExperimentalWarning: queueMicrotask() is experimental.
(node:3852) ExperimentalWarning: queueMicrotask() is experimental.
(node:3940) ExperimentalWarning: queueMicrotask() is experimental.
(node:3744) ExperimentalWarning: queueMicrotask() is experimental.
(node:3772) ExperimentalWarning: queueMicrotask() is experimental.
(node:3700) ExperimentalWarning: queueMicrotask() is experimental.
(node:3896) ExperimentalWarning: queueMicrotask() is experimental.
(node:3684) ExperimentalWarning: queueMicrotask() is experimental.
(node:3724) ExperimentalWarning: queueMicrotask() is experimental.
(node:3784) ExperimentalWarning: queueMicrotask() is experimental.
(node:3736) ExperimentalWarning: queueMicrotask() is experimental.
(node:3696) ExperimentalWarning: queueMicrotask() is experimental.
(node:3708) ExperimentalWarning: queueMicrotask() is experimental.
(node:3644) ExperimentalWarning: queueMicrotask() is experimental.
(node:3588) ExperimentalWarning: queueMicrotask() is experimental.
(node:3672) ExperimentalWarning: queueMicrotask() is experimental.
(node:3780) ExperimentalWarning: queueMicrotask() is experimental.
(node:3676) ExperimentalWarning: queueMicrotask() is experimental.
(node:3668) ExperimentalWarning: queueMicrotask() is experimental.
(node:3784) ExperimentalWarning: queueMicrotask() is experimental.
(node:3828) ExperimentalWarning: queueMicrotask() is experimental.
(node:3816) ExperimentalWarning: queueMicrotask() is experimental.
(node:3792) ExperimentalWarning: queueMicrotask() is experimental.
(node:5708) ExperimentalWarning: queueMicrotask() is experimental.
(node:3724) ExperimentalWarning: queueMicrotask() is experimental.
(node:7136) ExperimentalWarning: queueMicrotask() is experimental.
(node:884) ExperimentalWarning: queueMicrotask() is experimental.
(node:3764) ExperimentalWarning: queueMicrotask() is experimental.
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

**Comment by tradexsrl** [comment]
i have run the exe from the command line with --debug but this is the output: 
![immagine](https://user-images.githubusercontent.com/62424179/150746037-3133d744-1474-4b9c-bde1-381516016fe3.png)

**Comment by JSuenram** [comment]
Sorry, not using on Windows, please try without "--debug" and/or/also with "-debug" only

**Comment by tradexsrl** [comment]
as you can see there is no -debug and all option are without -

**Comment by JSuenram** [comment]
> as you can see there is no -debug and all option are without -

Yes, just run without any parameter or with "test"....

**Comment by tradexsrl** [comment]
running with test it doesn't show nothing , running with only meshcentral.exe it gives this (sorry is in italian)
![immagine](https://user-images.githubusercontent.com/62424179/150759411-2931f5ed-8521-4c0d-bad4-b37fb7758e56.png)

**Comment by JSuenram** [comment]
Sorry,.... need to read this from the manual for Windows-Installation, as we are only using Debians here for MC.
Please try to run MeshCentral from Commandline by going where your NODE-JS-Folder is and try running MeshCentral with this command.
node node_modules/meshcentral

Taken from https://info.meshcentral.com/downloads/MeshCentral2/MeshCentral2InstallGuide.pdf

**Comment by tradexsrl** [comment]
...i think with that command i get more trouble now....please see the output:
![immagine](https://user-images.githubusercontent.com/62424179/150766274-5d8fe1b0-04e3-41c1-9771-8b009f6079f4.png)

**Comment by tradexsrl** [comment]
i revert back to version 0.9.72 (thank's to Veeam backup which is everytime a lifeline) and now the sever is back on track
i think version 0.9.73 which was autoinstalled on january 24th at midninght is completely bugged.
i have disable autoupdate feature

**Comment by JSuenram** [comment]
This can happen if you run on development-build. Latest stable is 0.9.52....
![image](https://user-images.githubusercontent.com/68963439/150772666-4a384bfb-7079-430a-b4a1-d48a92803d41.png)

But I can also confirm on Debian 0.9.7.3 has no problems so far.

The LOG from Command-Line looks like it is missing some modules on your installation. It tries to Install them but is Missing or unable to Load the "node-windows" module.... not sure why.

**Comment by tradexsrl** [comment]
i hope @Ylianst can shed some lights on this problem....

**Comment by tradexsrl** [comment]
hi , thank's for the technical explanation.
hard disk space is plenty (free 14gb)
i have checked into that path (\npm-cache\...) and i have found a log but from 2020 from the property of file...when i do the restore i just restored only the directory meshcentral into c:\program files\open source\ so the state of the file into \roaming\ was not changed.
anyway this is the log:
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'rebuild' ]
2 info using npm@6.7.0
3 info using node@v11.10.1
4 verbose run-script [ 'prerebuild', 'rebuild', 'postrebuild' ]
5 info lifecycle ursa-optional@0.9.10~prerebuild: ursa-optional@0.9.10
6 info lifecycle ursa-optional@0.9.10~rebuild: ursa-optional@0.9.10
7 verbose lifecycle ursa-optional@0.9.10~rebuild: unsafe-perm in lifecycle true
8 verbose lifecycle ursa-optional@0.9.10~rebuild: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Program Files\Open Source\MeshCentral\node_modules\ursa-optional\node_modules\.bin;C:\Program Files\Open Source\MeshCentral\node_modules\.bin;C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Program Files\Open Source\MeshCentral\node_modules\ursa-optional\node_modules\.bin;C:\Program Files\Open Source\MeshCentral\node_modules\.bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\nodejs\;C:\Users\administrator\AppData\Local\Microsoft\WindowsApps;C:\Users\administrator\AppData\Roaming\npm
9 verbose lifecycle ursa-optional@0.9.10~rebuild: CWD: C:\Program Files\Open Source\MeshCentral\node_modules\ursa-optional
10 silly lifecycle ursa-optional@0.9.10~rebuild: Args: [ '/d /s /c', 'node-gyp rebuild' ]
11 silly lifecycle ursa-optional@0.9.10~rebuild: Returned: code: 1  signal: null
12 info lifecycle ursa-optional@0.9.10~rebuild: Failed to exec rebuild script
13 verbose stack Error: ursa-optional@0.9.10 rebuild: `node-gyp rebuild`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:197:13)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:197:13)
13 verbose stack     at maybeClose (internal/child_process.js:984:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:265:5)
14 verbose pkgid ursa-optional@0.9.10
15 verbose cwd C:\Program Files\Open Source\MeshCentral\node_modules\ursa-optional
16 verbose Windows_NT 10.0.18363
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "rebuild"
18 verbose node v11.10.1
19 verbose npm  v6.7.0
20 error code ELIFECYCLE
21 error errno 1
22 error ursa-optional@0.9.10 rebuild: `node-gyp rebuild`
22 error Exit status 1
23 error Failed at the ursa-optional@0.9.10 rebuild script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by Ylianst** [comment]
Make sure to backup a working copy of `node-modules` before doing any update. If anything goes wrong, you can just put the working `node_modules` folder back and start up the server again.

Also, you are on NodeJS v11 and see the following warnings:

```
(node:3928) ExperimentalWarning: queueMicrotask() is experimental.
```

Update to a newer version of NodeJS to fix these warnings.

Personally, I would rename `node_modules` to `node_modules_working' and install MeshCentral again manually using `npm install meshcental` then run manually using `node node_modules meshcentral`. That gives you a fresh copy of `node_modules` to all the latest dependencies.

**Comment by tradexsrl** [comment]
i do the command

> As has been pointed out, meshcentral was failing to load the module `node-windows`. This is a library that meshcentral is dependant on when being run as a windows service. It is not clear if the module was missing or broken. but it looks like it was missing.
> 
> When meshcentral updates itself, internally, it is actually just running `npm install meshcentral@latest` and restarting itself. This kind of error can happen if `npm install` fails, for example, if the server runs out of disk space during the installation.
> 
> npm will generally log its output during a failure. Usually, this can be found somewhere like `C:\Users\XXX\AppData\Roaming\npm-....\`
> 
> If you cannot find a reason from the npm logs (or you can't find the npm logs), you can try going to `C:\Program Files\Open Source\MeshCentral` and running `npm install meshcentral@latest` manually and see what happens.

i do the command npm install meshcentral@latest and again the same problem

**Comment by JSuenram** [comment]
Please try as Ylianst suggested.

`Personally, I would rename node_modules to node_modules_working' and install MeshCentral again manually using npm install meshcentalthen run manually usingnode node_modules meshcentral. That gives you a fresh copy of node_modules` to all the latest dependencies.`

**Comment by JSuenram** [comment]
Maybe there is an older Node-Version in the MeshCentral-Installation folder.

**Comment by tradexsrl** [comment]
> Maybe there is an older Node-Version in the MeshCentral-Installation folder.

i'm on windows and i downloaded nodejs installer lts and installed over the previous , it say it has upgraded to the latest , as node -v report
there is no other older version

**Comment by nzalev** [comment]
I remember you had opened #3484 a few days ago, I believe the microstack queue warnings are from a previous day and are probably unrelated.

It is interesting that updating to 0.9.73 would break loading `node-windows` - that part of meshcentral.js hasn't been changed in nearly a year, and `node-windows` itself hasn't had a new release in 4 months.

I will have to break out a windows system to test :)

**Comment by JSuenram** [comment]
> > Maybe there is an older Node-Version in the MeshCentral-Installation folder.
> 
> i'm on windows and i downloaded nodejs installer lts and installed over the previous , it say it has upgraded to the latest , as node -v report there is no other older version

Maybe, if you can't wait, backup your config and database-file and delete all MeshCentral-Installation-Folders and dependent Software and start from scratch step by step. But really make sure you have a backup of your config data.
Just an Idea.

**Comment by tradexsrl** [comment]
> Please try as Ylianst suggested.
> 
> Personally, I would rename node_modules to node_modules_working' and install MeshCentral again manually using npm install meshcentalthen run manually usingnode node_modules meshcentral. That gives you a fresh copy of `node_modules` to all the latest dependencies.`

i have done what suggested @Ylianst , i get this output:
![immagine](https://user-images.githubusercontent.com/62424179/150824656-a27e74b6-b5d4-4182-8e4c-e30ac25664a6.png)

**Comment by tradexsrl** [comment]
is it normal ?

**Comment by nzalev** [comment]
That is normal. You would now run meshcentral with `node node_modules/meshcentral`.

I am currently trying to replicate this on Windows - so far, everything is working fine for me. I am currently updating my node version to v16 to see if that is related.

**Comment by tradexsrl** [comment]
ok i do the command above but i get this:
![immagine](https://user-images.githubusercontent.com/62424179/150825370-59a77f47-c0cc-4345-8cfe-0c4e76cee819.png)

**Comment by tradexsrl** [comment]
aaaaaaaaaaah you have found the culprit.... nodejs v16 lts .... @Ylianst ask me to update from node v11 that comes preinstalled with meshcentral because of the warning i reported...with nodejs v16 warning goes away but other trouble start....i'll leave autoupdate disable until someone find what i need to do with nodejs version installed

**Comment by nzalev** [comment]
@tradexsrl

I am getting the following behaviour:
- When meshcentral is run the first time, it outputs "Installing node-windows@0.1.4..."
- Meshcentral fails to start with the error you see

After it installs the module, it seems to be OK.
- Meshcentral is run the second time
- it starts normally.

Can you do the following:
- run `node meshcentral` such that it displays "Installing node-windows@0.1.4..." and fails (as you shows in your screenshot)
- run `node meshcentral` again (a second time).

It is working consistently for me that way.

**Comment by tradexsrl** [comment]
very strange...now i redo all the sequence: rename the node_modules directory , then redo npm install meshcentral , then send the command: node meshcentral and it start successfully with version 0.9.73..verified also to restart the windows vm and everythings is working now...i suspect npm package was bugged or somethings else ...i see that in the old node_modules dir there is 715 dir , in the new one only 136 dir...for example in the new \node_modules\ not anymore dir starting with "le"
![immagine](https://user-images.githubusercontent.com/62424179/150839558-f5f8a35e-fb18-4e66-828e-7ad939fdb1fd.png)

i don't know if this could arise some other problem because i'm using Let's encrypt for certificate and for autorenew of the cert...

**Comment by tradexsrl** [comment]
So I'm safe with what I have now in the node_modules dir? My installation is very basic

**Comment by tradexsrl** [comment]
To mention that I'm running meshcentral for over 2 years with autoupdate feature setup to true without any problem ... only now after updating nodejs comes out this serious problem

**Comment by nzalev** [comment]
As long as it has already installed `node-windows`, you will be good to go.

I am not sure what caused meshcentral to begin trying to install a new copy of `node-windows` - it could have been a result of your updating Node.js, but I am not sure.

**Comment by tradexsrl** [comment]
I was thinking if it could be possible to implement this workaround if no culprit could be found : if the require is not able to find node-windows try only another time do meshcentral start

**Comment by tradexsrl** [comment]
> As long as it has already installed `node-windows`, you will be good to go.
> 
> 
> 
> I am not sure what caused meshcentral to begin trying to install a new copy of `node-windows` - it could have been a result of your updating Node.js, but I am not sure. 

Are you referring to npm command or the node command where it shows install node-windows ?
I do not check if when I run npm install it install also node-windows or not...if with npm command that dir is created is strange that node meshcentral command redo that ...if instead node-windows isn't there after npm command (because as suggested I rename it) as you write that dir is created at runtime and maybe is not yet ready when the server try to start...maybe with a little delay the problem goes away...but is only my thinking knowing nothing of node programming

**Comment by nzalev** [comment]
`npm install meshcentral` is not installing `node-windows` - it is installed while meshcentral is running. (this is because linux installs, for example, have no need for node-windows)

> maybe is not yet ready when the server try to start...maybe with a little delay the problem goes away

This is one of the first things I tested. I set meshcentral to launch 15s after the installation of node-windows to confirm that it fully installs first - no luck.

I think to fully understand why it fails, I will have to do some more research.

In the mean time though, when meshcentral outputs `Module node-windows can't be loaded. Restart Meshcentral.`, it is sufficient to simply re-run `node node_modules/meshcentral` and it should work.
